### PR TITLE
s_server: add option -noninteractive

### PR DESCRIPTION
**Pull request by cnt0**
This option prevents s_server from reading user input like -brief and -quiet do, but it doesn't suppress the output, which is useful for automated testing.

CLA: trivial

##### Checklist

- [x] documentation is added or updated

### CODE REVIEW

File: apps/s_server.c [hunk 1]
```diff
@@ -1413,6 +1417,8 @@ int s_server_main(int argc, char *argv[])
         case OPT_BRIEF:
             s_quiet = s_brief = verify_args.quiet = 1;
             break;
+        case OPT_NONINTERACTIVE:
+            s_noninteractive = 1;
         case OPT_NO_DHE:
```
Comment (slontis) [inline apps/s_server.c:1422]: break;

**Review by paulidale — COMMENTED** [review]
This is well beyond our definition of trivial.  A CLA will be required.

**Comment by cnt0** [comment]
> This is well beyond our definition of trivial. A CLA will be required.

CLA is no problem, I'll sign that if you'll decide to merge that PR. Should I remove "CLA: trivial" from commit messages?

File: apps/s_server.c (hunk 1 above)
Comment (cnt0) [reply apps/s_server.c:1423]: https://github.com/openssl/openssl/pull/16193/commits/88f198136ed8319ab917385c5f30677ea7207976

fixed, my bad

**Comment by t-j-h** [comment]
> This is well beyond our definition of trivial. A CLA will be required.

Agreed. It is simple - but it is not trivial. CLA required.

**Comment by paulidale** [comment]
> Should I remove "CLA: trivial" from commit messages?

Yes please.

**Comment by cnt0** [comment]
OK, so looks like we'll need a CCLA here. Prior to further steps we would be grateful if you supply us with your comments on the following questions:

- Is this agreement an offer or we can make amendments to it?
- Is this a fixed-term or open-ended agreement?
- Does the OpenSSL Software Foundation (the "Foundation") have any restrictions on cooperation with countries (where contributing companies are registered) under a CCL Agreement?

Thank you for your time and hope for bilateral mutually beneficial cooperation.

Kind regards,

**Comment by richsalz** [comment]
FYI, I would be surprised if the project accepted this PR for the 3.0 release, so there's no rush.  Note that I am not a member of the project, just an informed bystander.

**Comment by paulidale** [comment]
These are questions for a lawyer but I'll try to respond as I see things (i.e. my comments cannot override, alter or replace anything in either CLA).

The agreements grant the project licenses to use and to continue to use contributed code.  They are not intended to reduce or restrict any authors' rights to their work in any other way.


>     * Is this agreement an offer or we can make amendments to it?

We accept no alterations to the CLA terms.


>     * Is this a fixed-term or open-ended agreement?

The licenses granted are _perpetual_, so you can't contribute under the ICLA and then later withdraw permission for us to use your contribution.


>     * Does the OpenSSL Software Foundation (the "Foundation") have any restrictions on cooperation with countries (where contributing companies are registered) under a CCL Agreement?

I don't believe so.  We've contributors from all around the world Europe, USA, China, Russia, ... and haven't encountered anything I'm aware of.


As @richsalz notes, this isn't going to make 3.0, so there isn't a rush to sort out the CLA side.

**Comment by cnt0** [comment]
I changed the email in commit so it matches my work one. We'll sign all required CLAs soon.

Oh and congratulations with big release.

**Pull request closed without merging**